internal/provider/resource/secret_sync/secret_sync_render.go

```diff
@@ -0,0 +1,340 @@
+	if !exists {
+		diags.AddError("Invalid destination config", "Expected 'type' to be present")
+		return false
+	}
+
+	typeVal, ok := typeAttr.(types.String)
+	if !ok {
+		diags.AddError("Invalid destination config", "Expected 'type' to be a string type")
+		return false
+	}
+
+	if typeVal.IsNull() || typeVal.IsUnknown() {
+		diags.AddError("Invalid destination config", "Expected 'type' to have a value")
+		return false
+	}
+
+	syncType := RenderSyncType(typeVal.ValueString())
+
+	if syncType != RenderSyncTypeEnv && syncType != RenderSyncTypeFile {
+		diags.AddError("Invalid destination config", fmt.Sprintf("Invalid type '%s'.", syncType))
+	}
+
+	// Check required fields are not empty
+	for _, field := range requiredFields {
+		value, exists := destinationConfig[field]
+		if !exists {
+			diags.AddError("Invalid destination config", fmt.Sprintf("Expected '%s' to be present", field))
+			return false
+		}
+
+		// Check if the value is null, unknown, or empty based on its type
+		if terraform.IsAttrValueEmpty(value) {
+			diags.AddError("Invalid destination config", fmt.Sprintf("Expected '%s' to be set", field))
+			return false
+		}
+	}
+
+	// Check for unexpected fields
+	allowedFieldsMap := make(map[string]bool)
+	for _, field := range requiredFields {
+		allowedFieldsMap[field] = true
+	}
+
+	for field := range destinationConfig {
+		if !allowedFieldsMap[field] {
+			if terraform.IsAttrValueEmpty(destinationConfig[field]) {
+				continue
+			}
+
+			diags.AddError("Invalid destination config", fmt.Sprintf("Unexpected field '%s' for scope '%s'. Supported destination_config fields are: %v", field, scope, requiredFields))
+			return false
+		}
+	}
+
+	return true
+}
+
+// SecretSyncRenderDestinationConfigModel describes the data source data model.
+
+type SecretSyncRenderDestinationConfigModel struct {
+	Scope     types.String `tfsdk:"scope"` // service
+	Type      types.String `tfsdk:"type"`  // env
+	ServiceID types.String `tfsdk:"service_id"`
+}
+
+type SecretSyncRenderSyncOptionsModel struct {
+	InitialSyncBehavior   types.String `tfsdk:"initial_sync_behavior"`
+	DisableSecretDeletion types.Bool   `tfsdk:"disable_secret_deletion"`
+	KeySchema             types.String `tfsdk:"key_schema"`
+}
+
+func NewSecretSyncRenderResource() resource.Resource {
+	return &SecretSyncBaseResource{
+		App:              infisical.SecretSyncAppRender,
+		SyncName:         "Render",
+		ResourceTypeName: "_secret_sync_render",
+		AppConnection:    infisical.AppConnectionAppRender,
+		DestinationConfigAttributes: map[string]schema.Attribute{
+			"scope": schema.StringAttribute{
+				Required:    true,
+				Description: "The Render scope that secrets should be synced to.",
+			},
+			"service_id": schema.StringAttribute{
+				Required:    true,
+				Description: "The ID of the Render service to sync secrets to.",
+			},
+			"type": schema.StringAttribute{
+				Required:    true,
+				Description: "The Render resource type to sync secrets to.",
+			},
```

Comment:
please add the supported values for scope and type in the description so that it can be seen in the docs